### PR TITLE
ttyd: add test & use systemd LoadCredential

### PR DESCRIPTION
**Pull request by Stunkymonkey**
## Description of changes

part of #156400

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
just 10 min later, I am wonder if this is helping in any way.

I am too tired right now, will look at it tomorrow.

File: nixos/modules/services/web-servers/ttyd.nix [hunk 1]
```diff
@@ -64,8 +64,7 @@ in
         apply = value: if value == null then null else toString value;
         description = lib.mdDoc ''
           File containing the password to use for basic authentication.
-          For insecurely putting the password in the globally readable store use
-          `pkgs.writeText "ttydpw" "MyPassword"`.
+          For setting the password use `pkgs.writeText "ttydpw" "MyPassword"`.
```
Comment (NickCao) [inline nixos/modules/services/web-servers/ttyd.nix:67]: This is still insecure right? Hiding the store path is by no means a security mechanism.

Comment (Stunkymonkey) [reply nixos/modules/services/web-servers/ttyd.nix:67]: it totally is. Sorry not sure what I thought here.

Maybe we just keep the test and the rest gets removed?

Comment (NickCao) [reply nixos/modules/services/web-servers/ttyd.nix:67]: The use of `LoadCredential` might also be kept, only the `description` changes shall be reverted.

File: nixos/tests/web-servers/ttyd.nix [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import ../make-test-python.nix ({ lib, pkgs, ... }: {
+  name = "freshrss";
+  meta.maintainers = with lib.maintainers; [ etu stunkymonkey ];
```
Comment (NickCao) [inline nixos/tests/web-servers/ttyd.nix:3]: cc @etu

File: nixos/tests/web-servers/ttyd.nix [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      username = "foo";
+      passwordFile = pkgs.writeText "password" "bar";
+    };
+  };
+
+  testScript = ''
+    machine.wait_for_unit("multi-user.target")
+    machine.wait_for_open_port(7681)
+    response = machine.succeed("curl -vvv -u foo:bar -s -H 'Host: ttyd' http://127.0.0.1:7681/")
+    print(response[0:1000])
```
Comment (NickCao) [inline nixos/tests/web-servers/ttyd.nix:17]:
```suggestion
```

Suppose this is for debugging, please drop it.

File: nixos/tests/web-servers/ttyd.nix (hunk 2 above)
Comment (Stunkymonkey) [reply nixos/tests/web-servers/ttyd.nix:3]: sorry copy paste mistake

**Comment by Stunkymonkey** [comment]
really feel ashamed, how bad the state of the PR was, when requesting a review. Thanks for your patience.

**Comment by NickCao** [comment]
@ofborg eval

**Comment by trofi** [comment]
`ttyd.tests` failed the val for me for missing `nixosTests` import. Proposed the fix as https://github.com/NixOS/nixpkgs/pull/295448

**Pull request merged**